From ac6385f23d494bd82a4cb508d0e5164ee3e830b6 Mon Sep 17 00:00:00 2001
From: Kokan <kokaipeter@gmail.com>
Date: Thu, 21 Feb 2019 13:49:24 +0100
Subject: [PATCH] template: tf_simple_func_prepare leak fix

The state object had the array and the argc length, when the proper
cleanup function is called it checks the argc to free templates.

Signed-off-by: Kokan <kokaipeter@gmail.com>
---
 lib/template/simple-function.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/template/simple-function.c b/lib/template/simple-function.c
index 5f788aed63..ffaa7b7c41 100644
--- a/lib/template/simple-function.c
+++ b/lib/template/simple-function.c
@@ -55,7 +55,10 @@ tf_simple_func_prepare(LogTemplateFunction *self, gpointer s, LogTemplate *paren
       state->argv_templates[i] = log_template_new(parent->cfg, NULL);
       log_template_set_escape(state->argv_templates[i], parent->escape);
       if (!log_template_compile(state->argv_templates[i], argv[i + 1], error))
-        goto error;
+        {
+          state->argc = i + 1;
+          goto error;
+        }
     }
   state->argc = argc - 1;
   return TRUE;
